server/src/main/java/org/elasticsearch/index/query/QueryShardContext.java

```diff
@@ -241,7 +265,7 @@ public MappedFieldType fieldMapper(String name) {
         if (name.equals(TypeFieldMapper.NAME)) {
             deprecationLogger.deprecate("query_with_types", TYPES_DEPRECATION_MESSAGE);
         }
-        return failIfFieldMappingNotFound(name, mapperService.fieldType(name));
+        return failIfFieldMappingNotFound(name, lookupFieldType.apply(name));
```

Comment:
I am not sure I see the value in using a different function from mapper service. If the function is only used here, can't we have the logic here without a function and call the usual function for the ordinary mapped fields?

Reply:
I like the Function because it forces the parsing and lookup code close together and makes testing pretty easy. I'm certainly open to other ideas but I had the parsing here at first and didn't like it. I'd be happy to move the function's creation to a static method if you'd prefer that. That way I can still test it without building all the stuff to make a whole QueryShardContext.